### ISSUE TITLE
Thermostat card doesn't always trigger heater

### ISSUE THREAD

**Issue by benborra** [home-assistant/frontend#19402]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

This morning when adjusting the temperature the heater wouldn't turn on. Had to toggle the temperature down and up again to get it to kick in.
This is the second time I've had this occur.
Not that I'm not sliding the thermostat button up, but rather clicking on the desired target temperature.
I haven't encountered this with sliding or using the + and - buttons.

![HA](https://github.com/home-assistant/frontend/assets/18417334/b79cbe93-3610-4ed6-8719-2134e1e6aa06)

I made the above recording to indicate that there seems to be a delay in the heater kicking in, but I just now noticed that the original temperature I set was max, doesn't seem to be reflected by the history. Only when toggling it down and back up.

### Describe the behavior you expected

Clicking the desired temperature reflects in the system what is shown in the UI

### Steps to reproduce the issue

1. Open app (android)
2. Click in the thermostat on the temperature you want
3. Sometimes the heater doesn't kick in.

### What version of Home Assistant Core has the issue?

2024.1.3

### What was the last working version of Home Assistant Core?

Not sure

### In which browser are you experiencing the issue with?

2023.12.4-full

### Which operating system are you using to run this browser?

ColorOS 13.1

### Problem-relevant frontend configuration

```yaml
- platform: generic_thermostat
  name: Badkamer
  heater: switch.stopcontact_verwarming_badkamer_switch
  target_sensor: sensor.lumi_lumi_sens_temperature_2
  min_temp: 8
  max_temp: 25
  ac_mode: false
  target_temp: 10
  cold_tolerance: 0.2
  hot_tolerance: 0
  min_cycle_duration:
    seconds: 5
  keep_alive:
    minutes: 3
  initial_hvac_mode: "heat"
  away_temp: 10
  precision: 1
  unique_id: Thermostat.Badkamer
```

**Comment by karwosts** [comment]
Cards do not trigger heaters. The only thing the card does is change the setpoints (via service calls), and change the modes (via service calls).

If the setpoint succesfully changed, but the heater did not come on, that's a backend problem with the integration, not with the card. 

If the setpoint did not succesfully change, than that is something that could be looked at here.

**Comment by benborra** [comment]
How can I validate if it's the setpoint that didn't change?
Or is that indicated by the fact that the target in the history didn't change?

Couldn't it be that at a certain point, the command didn't go through?

**Comment by karwosts** [comment]
You can validate the setpoints with either developer tools or a history graph card. 

Yes its theoretically possible a command could be dropped somewhere.

**Comment by benborra** [comment]
Got it on recording. I can quite easily trigger it by tapping the temperature back and forth a couple of times.
https://i.imgur.com/9qiMbSa.mp4

It seems that only the 3rd tap was registered, you'll notice that the card didn't switch to 'heating' and when I opened the details view, it was still at it's original 10°C.

Note that I haven't had any problems with sliding it.

**Comment by karwosts** [comment]
Is this only a mobile thing? If you have it on desktop I would say you can open two browser tabs, and in the second tab you can open devtools->events and listen for `call_service` events. 

Then you should see what service the frontend calls each time you tap the card, so you can see if it's making the correct calls to change the setpoints.

**Comment by benborra** [comment]
It only seems to occur on my phone, both through the app, and through browser (tested with firefox).
I subscribed to the call_service events and also didn't notice a new event arriving when the issue occured, where they did for all the others taps.

I also just noticed, when you tap on a temperature target, there seems to be a small animation of the dial turning.
However, when the error occurs, the animation doesn't happen. I just instantly goes to the location that I clicked.
